#! /bin/sh /usr/share/dpatch/dpatch-run
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Backport helper function needed for security fixes from 1.0.4 <jmm@debian.org>

@DPATCH@
diff -aur wireshark-0.99.4.orig/epan/packet.c wireshark-0.99.4/epan/packet.c
--- wireshark-0.99.4.orig/epan/packet.c	2008-07-10 19:39:28.000000000 +0200
+++ wireshark-0.99.4/epan/packet.c	2008-11-14 00:34:02.000000000 +0100
@@ -1865,6 +1865,24 @@
     num_of_postdissectors++;
 }
 
+gboolean
+have_postdissector()
+{
+    guint i;
+    dissector_handle_t handle;
+
+    for(i = 0; i < num_of_postdissectors; i++) {
+	handle = (dissector_handle_t) g_ptr_array_index(post_dissectors,i);
+
+	if (handle->protocol != NULL
+	    && proto_is_protocol_enabled(handle->protocol)) {
+	    /* We have at least one enabled postdissector */
+	    return TRUE;
+	}
+    }
+    return FALSE;
+}
+
 extern void call_all_postdissectors(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree) {
     guint i;
     for(i=0;i<num_of_postdissectors;i++) {
diff -aur wireshark-0.99.4.orig/epan/packet.h wireshark-0.99.4/epan/packet.h
--- wireshark-0.99.4.orig/epan/packet.h	2008-07-10 19:39:28.000000000 +0200
+++ wireshark-0.99.4/epan/packet.h	2008-11-14 00:34:27.000000000 +0100
@@ -413,6 +413,7 @@
  * dissector has been called.
  */
 extern void register_postdissector(dissector_handle_t);
+extern gboolean have_postdissector();
 extern void call_all_postdissectors(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree);
 
 #ifdef __cplusplus
